pint_xarray/tests/test_accessors.py

```diff
@@ -86,13 +86,27 @@ def test_attach_units_from_attrs(self, example_unitless_da):
         remaining_attrs = conversion.extract_unit_attributes(result)
         assert {k: v for k, v in remaining_attrs.items() if v is not None} == {}
 
+    def test_attach_units_from_str_attr_no_unit(self, example_unitless_da):
+        orig = example_unitless_da
+        orig.attrs["units"] = "none"
+        result = orig.pint.quantify("m")
+        assert_array_equal(result.data.magnitude, orig.data)
+        assert str(result.data.units) == "meter"
+
     def test_attach_units_given_unit_objs(self, example_unitless_da):
         orig = example_unitless_da
         ureg = UnitRegistry(force_ndarray=True)
         result = orig.pint.quantify(ureg.Unit("m"), unit_registry=ureg)
         assert_array_equal(result.data.magnitude, orig.data)
         assert result.data.units == ureg.Unit("m")
 
+    @pytest.mark.parametrize("no_unit_value", conversion.no_unit_values)
+    def test_override_units(self, example_unitless_da, no_unit_value):
+        orig = example_unitless_da
+        result = orig.pint.quantify(no_unit_value, u=no_unit_value)
+        assert getattr(result.data, "units", "not a quantity") == "not a quantity"
+        assert getattr(result["u"].data, "units", "not a quantity") == "not a quantity"
```

Comment:
This seems a little unconventional - wouldn't it be better to check for the expected error via

```python
result_data, result_u_data = result.data, result["u"].data
with pytest.raises(AttributeError):
    result_data.units
with pytest.raises(AttributeError):
    result_u_data.units
```

Reply:
you're absolutely right, I must have been way too tired when I wrote this. I'll change it.